module/VuFind/src/VuFind/Controller/CommentsController.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * Comments Controller
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Controller;
+
+use VuFind\Exception\Forbidden as ForbiddenException;
+use VuFind\Validator\CsrfInterface;
+
+/**
+ * Comments controller.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class CommentsController extends AbstractBase
+{
+    use Feature\UserContentTrait;
+
+    /**
+     * Array of sort options for userListAction
+     *
+     * @var array
+     */
+    protected array $sortList = [
+        'created desc' => 'sort_created_desc',
+        'created asc' => 'sort_created_asc',
+        'title' => 'sort_title',
+    ];
+
+    /**
+     * Get all comments for the logged in user
+     *
+     * @return View
+     */
+    public function userListAction()
+    {
+        $user = $this->getUser();
+        if (!$user) {
+            return $this->forceLogin();
+        }
+        if (!$this->commentsEnabled()) {
+            throw new ForbiddenException('Comments disabled.');
+        }
+        $limit = $this->getService(\VuFind\Config\AccountCapabilities::class)->getUserContentPageSize();
+        $page = $this->params()->fromQuery('page', 1);
+        $sort = $this->params()->fromQuery('sort', 'created desc');
```

Comment:
We should probably do some sort of validation on sort, here and/or in the service itself. (I think it makes the most sense to validate here, so that if we add the ability to disable sort options in the future, the limit cannot be circumvented by modifying the URL directly).

Reply:
Might also make sense to validate that $page is a positive integer as well.

Reply:
Some validation added and also moved these to UserContentTrait